<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 5 katG Mutation Analysis (Isoniazid Resistance) | Tutorial for R qMut package (v1.0.0)</title>
  <meta name="description" content="Chapter 5 katG Mutation Analysis (Isoniazid Resistance) | Tutorial for R qMut package (v1.0.0)" />
  <meta name="generator" content="bookdown 0.43 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 5 katG Mutation Analysis (Isoniazid Resistance) | Tutorial for R qMut package (v1.0.0)" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 5 katG Mutation Analysis (Isoniazid Resistance) | Tutorial for R qMut package (v1.0.0)" />
  
  
  

<meta name="author" content="Hongxiang Xu" />


<meta name="date" content="2025-06-20" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="mutation-screening-and-analysis.html"/>
<link rel="next" href="multi-omics-integration-for-discovery-of-non-neutral-mutations.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>
<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Tutorial for R qMut package</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> About</a></li>
<li class="chapter" data-level="2" data-path="installation.html"><a href="installation.html"><i class="fa fa-check"></i><b>2</b> <strong>Installation</strong></a></li>
<li class="chapter" data-level="3" data-path="data-integration.html"><a href="data-integration.html"><i class="fa fa-check"></i><b>3</b> Data Integration</a>
<ul>
<li class="chapter" data-level="3.1" data-path="data-integration.html"><a href="data-integration.html#input-requirements"><i class="fa fa-check"></i><b>3.1</b> Input Requirements</a></li>
<li class="chapter" data-level="3.2" data-path="data-integration.html"><a href="data-integration.html#creating-the-snp-object"><i class="fa fa-check"></i><b>3.2</b> Creating the SNP Object</a></li>
<li class="chapter" data-level="3.3" data-path="data-integration.html"><a href="data-integration.html#merging-objects-for-expanded-analysis"><i class="fa fa-check"></i><b>3.3</b> Merging Objects for Expanded Analysis</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="mutation-screening-and-analysis.html"><a href="mutation-screening-and-analysis.html"><i class="fa fa-check"></i><b>4</b> Mutation Screening and Analysis</a>
<ul>
<li class="chapter" data-level="4.1" data-path="mutation-screening-and-analysis.html"><a href="mutation-screening-and-analysis.html#basic-mutation-statistics"><i class="fa fa-check"></i><b>4.1</b> Basic Mutation Statistics</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="katg-mutation-analysis-isoniazid-resistance.html"><a href="katg-mutation-analysis-isoniazid-resistance.html"><i class="fa fa-check"></i><b>5</b> katG Mutation Analysis (Isoniazid Resistance)</a></li>
<li class="chapter" data-level="6" data-path="multi-omics-integration-for-discovery-of-non-neutral-mutations.html"><a href="multi-omics-integration-for-discovery-of-non-neutral-mutations.html"><i class="fa fa-check"></i><b>6</b> Multi-omics Integration for Discovery of Non-neutral Mutations</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Tutorial for R qMut package (v1.0.0)</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="katg-mutation-analysis-isoniazid-resistance" class="section level1 hasAnchor" number="5">
<h1><span class="header-section-number">Chapter 5</span> katG Mutation Analysis (Isoniazid Resistance)<a href="katg-mutation-analysis-isoniazid-resistance.html#katg-mutation-analysis-isoniazid-resistance" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="katg-mutation-analysis-isoniazid-resistance.html#cb26-1" tabindex="-1"></a><span class="co"># Identify all katG mutations</span></span>
<span id="cb26-2"><a href="katg-mutation-analysis-isoniazid-resistance.html#cb26-2" tabindex="-1"></a>katG_mut <span class="ot">&lt;-</span> Mtb_mut<span class="sc">$</span><span class="fu">estimate_imp</span>(<span class="st">&quot;katG&quot;</span>, <span class="at">gene_type =</span> <span class="st">&quot;GENE&quot;</span>)</span></code></pre></div>
<pre><code>## filter genes</code></pre>
<pre><code>## remain 51181 sample and 1271 snps</code></pre>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="katg-mutation-analysis-isoniazid-resistance.html#cb29-1" tabindex="-1"></a><span class="co"># Filter strains with R463L variant (a common INH resistance mutation)</span></span>
<span id="cb29-2"><a href="katg-mutation-analysis-isoniazid-resistance.html#cb29-2" tabindex="-1"></a>katG_R463L_index <span class="ot">&lt;-</span> katG_mut <span class="sc">%&gt;%</span> </span>
<span id="cb29-3"><a href="katg-mutation-analysis-isoniazid-resistance.html#cb29-3" tabindex="-1"></a>  <span class="fu">filter</span>(AA_effect_short <span class="sc">==</span> <span class="st">&quot;R463L&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb29-4"><a href="katg-mutation-analysis-isoniazid-resistance.html#cb29-4" tabindex="-1"></a>  <span class="fu">pull</span>(INDEX)</span>
<span id="cb29-5"><a href="katg-mutation-analysis-isoniazid-resistance.html#cb29-5" tabindex="-1"></a></span>
<span id="cb29-6"><a href="katg-mutation-analysis-isoniazid-resistance.html#cb29-6" tabindex="-1"></a><span class="co"># filter_sample function will keep samples with katG_R463L mutation.</span></span>
<span id="cb29-7"><a href="katg-mutation-analysis-isoniazid-resistance.html#cb29-7" tabindex="-1"></a>katG_R463L <span class="ot">&lt;-</span> Mtb_mut<span class="sc">$</span><span class="fu">filter_sample</span>(<span class="at">mutation_index =</span> katG_R463L_index)</span></code></pre></div>
<pre><code>## remain 24610 sample and 1293186 snps</code></pre>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="katg-mutation-analysis-isoniazid-resistance.html#cb31-1" tabindex="-1"></a><span class="co"># To examine how katG mutations are distributed across different bacterial </span></span>
<span id="cb31-2"><a href="katg-mutation-analysis-isoniazid-resistance.html#cb31-2" tabindex="-1"></a><span class="co"># lineages, we can perform lineage-specific frequency calculations </span></span>
<span id="cb31-3"><a href="katg-mutation-analysis-isoniazid-resistance.html#cb31-3" tabindex="-1"></a><span class="co"># statistically:</span></span>
<span id="cb31-4"><a href="katg-mutation-analysis-isoniazid-resistance.html#cb31-4" tabindex="-1"></a>Mtb_mut<span class="sc">$</span><span class="fu">estimate_imp</span>(<span class="st">&quot;katG&quot;</span>,<span class="at">gene_type =</span> <span class="st">&quot;GENE&quot;</span>,<span class="at">metadata_col =</span> <span class="st">&quot;Lineage&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb31-5"><a href="katg-mutation-analysis-isoniazid-resistance.html#cb31-5" tabindex="-1"></a>  <span class="fu">filter</span>(INDEX <span class="sc">%in%</span> katG_R463L_index)</span></code></pre></div>
<pre><code>## filter genes</code></pre>
<pre><code>## remain 51181 sample and 1271 snps</code></pre>
<pre><code>## calculate mutation frequency</code></pre>
<pre><code>##                          CHROM     POS    TYPE FTYPE STRAND    NT_POS  AA_POS
## C2154731Asnp         NC_018143 2154731     snp   CDS      - 1388/2223 463/740
## CGG2154731AGTcomplex NC_018143 2154731 complex   CDS      - 1388/2223 462/740
## CG2154731AAcomplex   NC_018143 2154731 complex   CDS      - 1388/2223 463/740
## CGG2154731AGAcomplex NC_018143 2154731 complex   CDS      - 1388/2223 462/740
##                                                                    EFFECT
## C2154731Asnp                       missense_variant c.1388G&gt;T p.Arg463Leu
## CGG2154731AGTcomplex missense_variant c.1386_1388delCCGinsACT p.Arg463Leu
## CG2154731AAcomplex     missense_variant c.1387_1388delCGinsTT p.Arg463Leu
## CGG2154731AGAcomplex missense_variant c.1386_1388delCCGinsTCT p.Arg463Leu
##                      LOCUS_TAG_NEW GENE                 PRODUCT AA_effect_short
## C2154731Asnp          RVBD_RS10020 katG catalase/peroxidase HPI           R463L
## CGG2154731AGTcomplex  RVBD_RS10020 katG catalase/peroxidase HPI           R463L
## CG2154731AAcomplex    RVBD_RS10020 katG catalase/peroxidase HPI           R463L
## CGG2154731AGAcomplex  RVBD_RS10020 katG catalase/peroxidase HPI           R463L
##                      LOCUS_TAG                INDEX percents_L2 percents_L4
## C2154731Asnp           Rv1908c         C2154731Asnp     0.98083      0.0021
## CGG2154731AGTcomplex   Rv1908c CGG2154731AGTcomplex     0.00007      0.0000
## CG2154731AAcomplex     Rv1908c   CG2154731AAcomplex     0.00007      0.0000
## CGG2154731AGAcomplex   Rv1908c CGG2154731AGAcomplex     0.00007      0.0000
##                      percents_L1 percents_L3 percents_L5 percents_L6
## C2154731Asnp             0.98321      0.9801     0.99448     0.98795
## CGG2154731AGTcomplex     0.00000      0.0000     0.00000     0.00000
## CG2154731AAcomplex       0.00000      0.0000     0.00000     0.00000
## CGG2154731AGAcomplex     0.00000      0.0000     0.00000     0.00000
##                      percents_LBOV percents_L7
## C2154731Asnp                 0.995     0.98113
## CGG2154731AGTcomplex         0.000     0.00000
## CG2154731AAcomplex           0.000     0.00000
## CGG2154731AGAcomplex         0.000     0.00000</code></pre>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="katg-mutation-analysis-isoniazid-resistance.html#cb36-1" tabindex="-1"></a><span class="co"># Analysis of Drug Resistance Mutation Co-occurrence Patterns</span></span>
<span id="cb36-2"><a href="katg-mutation-analysis-isoniazid-resistance.html#cb36-2" tabindex="-1"></a><span class="co"># Comparative Analysis of embB Mutations in katG R463L Strains vs. WT Strains</span></span>
<span id="cb36-3"><a href="katg-mutation-analysis-isoniazid-resistance.html#cb36-3" tabindex="-1"></a>embB_compare <span class="ot">&lt;-</span> Mtb_mut<span class="sc">$</span><span class="fu">estimate_imp</span>(<span class="st">&quot;embB&quot;</span>,<span class="at">gene_type =</span> <span class="st">&quot;GENE&quot;</span>,<span class="at">metadata_col =</span> katG_R463L_index[<span class="dv">1</span>])</span></code></pre></div>
<pre><code>## filter genes</code></pre>
<pre><code>## remain 51181 sample and 1882 snps</code></pre>
<pre><code>## calculate mutation frequency</code></pre>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="katg-mutation-analysis-isoniazid-resistance.html#cb40-1" tabindex="-1"></a><span class="co"># Display significant non-synonymous embB mutations</span></span>
<span id="cb40-2"><a href="katg-mutation-analysis-isoniazid-resistance.html#cb40-2" tabindex="-1"></a>embB_compare <span class="sc">%&gt;%</span> </span>
<span id="cb40-3"><a href="katg-mutation-analysis-isoniazid-resistance.html#cb40-3" tabindex="-1"></a>  <span class="fu">arrange</span>(FDR) <span class="sc">%&gt;%</span> </span>
<span id="cb40-4"><a href="katg-mutation-analysis-isoniazid-resistance.html#cb40-4" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">str_detect</span>(EFFECT, <span class="st">&quot;synonymous_variant&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb40-5"><a href="katg-mutation-analysis-isoniazid-resistance.html#cb40-5" tabindex="-1"></a>  <span class="fu">head</span>()</span></code></pre></div>
<pre><code>##                  CHROM     POS TYPE FTYPE STRAND    NT_POS    AA_POS
## A4247824Csnp NC_018143 4247824  snp   CDS      + 1133/3297  378/1098
## A4246729Gsnp NC_018143 4246729  snp   CDS      +   38/3297   13/1098
## A4247607Gsnp NC_018143 4247607  snp   CDS      +  916/3297  306/1098
## T4247580Gsnp NC_018143 4247580  snp   CDS      +  889/3297  297/1098
## A4249935Gsnp NC_018143 4249935  snp   CDS      + 3244/3297 1082/1098
## T4247691Csnp NC_018143 4247691  snp   CDS      + 1000/3297  334/1098
##                                               EFFECT LOCUS_TAG_NEW GENE
## A4247824Csnp  missense_variant c.1133A&gt;C p.Glu378Ala  RVBD_RS20020 embB
## A4246729Gsnp     missense_variant c.38A&gt;G p.Asn13Ser  RVBD_RS20020 embB
## A4247607Gsnp   missense_variant c.916A&gt;G p.Met306Val  RVBD_RS20020 embB
## T4247580Gsnp   missense_variant c.889T&gt;G p.Ser297Ala  RVBD_RS20020 embB
## A4249935Gsnp missense_variant c.3244A&gt;G p.Thr1082Ala  RVBD_RS20020 embB
## T4247691Csnp  missense_variant c.1000T&gt;C p.Tyr334His  RVBD_RS20020 embB
##                                 PRODUCT AA_effect_short LOCUS_TAG        INDEX
## A4247824Csnp arabinosyltransferase EmbB           E378A    Rv3795 A4247824Csnp
## A4246729Gsnp arabinosyltransferase EmbB            N13S    Rv3795 A4246729Gsnp
## A4247607Gsnp arabinosyltransferase EmbB           M306V    Rv3795 A4247607Gsnp
## T4247580Gsnp arabinosyltransferase EmbB           S297A    Rv3795 T4247580Gsnp
## A4249935Gsnp arabinosyltransferase EmbB          T1082A    Rv3795 A4249935Gsnp
## T4247691Csnp arabinosyltransferase EmbB           Y334H    Rv3795 T4247691Csnp
##              count_total count_1 count_0    n1    n0       P_value
## A4247824Csnp        5807    5753      54 24610 26571  0.000000e+00
## A4246729Gsnp         645     639       6 24610 26571 5.938948e-150
## A4247607Gsnp        3530    2328    1202 24610 26571 1.993314e-107
## T4247580Gsnp         377       6     371 24610 26571  1.696397e-73
## A4249935Gsnp         200     198       2 24610 26571  3.926395e-60
## T4247691Csnp         165     163       2 24610 26571  4.117553e-49
##                        FDR
## A4247824Csnp  0.000000e+00
## A4246729Gsnp 1.397138e-147
## A4247607Gsnp 4.168242e-105
## T4247580Gsnp  3.192620e-71
## A4249935Gsnp  6.717705e-58
## T4247691Csnp  5.535168e-47</code></pre>
<p>The analysis reveals that the mutation of <em>embB</em> E306V appears at significantly higher frequency in <em>katG</em> R463L strains, which may reflect selection pressure of combination therapy, as INH and ethambutol (EMB) are frequently co-administered.</p>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="katg-mutation-analysis-isoniazid-resistance.html#cb42-1" tabindex="-1"></a><span class="co"># Generate and visualize mutation frequency tree of embB mutations in katG R463L strains and compare lineage distribution</span></span>
<span id="cb42-2"><a href="katg-mutation-analysis-isoniazid-resistance.html#cb42-2" tabindex="-1"></a>mut_tree <span class="ot">&lt;-</span> katG_R463L<span class="sc">$</span><span class="fu">MutTree</span>(</span>
<span id="cb42-3"><a href="katg-mutation-analysis-isoniazid-resistance.html#cb42-3" tabindex="-1"></a>  <span class="at">locus_tag =</span> <span class="st">&quot;Rv3795&quot;</span>,  <span class="co"># embB gene locus</span></span>
<span id="cb42-4"><a href="katg-mutation-analysis-isoniazid-resistance.html#cb42-4" tabindex="-1"></a>  <span class="at">limit =</span> <span class="dv">10</span>,            <span class="co"># Display top 10 mutations</span></span>
<span id="cb42-5"><a href="katg-mutation-analysis-isoniazid-resistance.html#cb42-5" tabindex="-1"></a>  <span class="at">metadata_col =</span> <span class="st">&quot;Lineage&quot;</span> <span class="co"># compare lineage distribution</span></span>
<span id="cb42-6"><a href="katg-mutation-analysis-isoniazid-resistance.html#cb42-6" tabindex="-1"></a>)</span></code></pre></div>
<pre><code>## remain 24610 sample and 1534 snps
## remain 729 significant snps
## remain 61 samples with Freq &gt;= 10</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-11-1.png" width="672" /></p>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="mutation-screening-and-analysis.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="multi-omics-integration-for-discovery-of-non-neutral-mutations.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
  "sharing": {
    "github": false,
    "facebook": true,
    "twitter": true,
    "linkedin": false,
    "weibo": false,
    "instapaper": false,
    "vk": false,
    "whatsapp": false,
    "all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
  },
  "fontsettings": {
    "theme": "white",
    "family": "sans",
    "size": 2
  },
  "edit": {
    "link": "https://github.com/USERNAME/REPO/edit/BRANCH/index.Rmd",
    "text": "Edit"
  },
  "history": {
    "link": null,
    "text": null
  },
  "view": {
    "link": null,
    "text": null
  },
  "download": ["_main.pdf", "_main.epub"],
  "search": {
    "engine": "fuse",
    "options": null
  },
  "toc": {
    "collapse": "subsection"
  }
});
});
</script>

</body>

</html>
