<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 6 Multi-omics Integration for Discovery of Non-neutral Mutations | Tutorial for R qMut package (v1.0.0)</title>
  <meta name="description" content="Chapter 6 Multi-omics Integration for Discovery of Non-neutral Mutations | Tutorial for R qMut package (v1.0.0)" />
  <meta name="generator" content="bookdown 0.43 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 6 Multi-omics Integration for Discovery of Non-neutral Mutations | Tutorial for R qMut package (v1.0.0)" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 6 Multi-omics Integration for Discovery of Non-neutral Mutations | Tutorial for R qMut package (v1.0.0)" />
  
  
  

<meta name="author" content="Hongxiang Xu" />


<meta name="date" content="2025-06-20" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="katg-mutation-analysis-isoniazid-resistance.html"/>

<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>
<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Tutorial for R qMut package</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> About</a></li>
<li class="chapter" data-level="2" data-path="installation.html"><a href="installation.html"><i class="fa fa-check"></i><b>2</b> <strong>Installation</strong></a></li>
<li class="chapter" data-level="3" data-path="data-integration.html"><a href="data-integration.html"><i class="fa fa-check"></i><b>3</b> Data Integration</a>
<ul>
<li class="chapter" data-level="3.1" data-path="data-integration.html"><a href="data-integration.html#input-requirements"><i class="fa fa-check"></i><b>3.1</b> Input Requirements</a></li>
<li class="chapter" data-level="3.2" data-path="data-integration.html"><a href="data-integration.html#creating-the-snp-object"><i class="fa fa-check"></i><b>3.2</b> Creating the SNP Object</a></li>
<li class="chapter" data-level="3.3" data-path="data-integration.html"><a href="data-integration.html#merging-objects-for-expanded-analysis"><i class="fa fa-check"></i><b>3.3</b> Merging Objects for Expanded Analysis</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="mutation-screening-and-analysis.html"><a href="mutation-screening-and-analysis.html"><i class="fa fa-check"></i><b>4</b> Mutation Screening and Analysis</a>
<ul>
<li class="chapter" data-level="4.1" data-path="mutation-screening-and-analysis.html"><a href="mutation-screening-and-analysis.html#basic-mutation-statistics"><i class="fa fa-check"></i><b>4.1</b> Basic Mutation Statistics</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="katg-mutation-analysis-isoniazid-resistance.html"><a href="katg-mutation-analysis-isoniazid-resistance.html"><i class="fa fa-check"></i><b>5</b> katG Mutation Analysis (Isoniazid Resistance)</a></li>
<li class="chapter" data-level="6" data-path="multi-omics-integration-for-discovery-of-non-neutral-mutations.html"><a href="multi-omics-integration-for-discovery-of-non-neutral-mutations.html"><i class="fa fa-check"></i><b>6</b> Multi-omics Integration for Discovery of Non-neutral Mutations</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Tutorial for R qMut package (v1.0.0)</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="multi-omics-integration-for-discovery-of-non-neutral-mutations" class="section level1 hasAnchor" number="6">
<h1><span class="header-section-number">Chapter 6</span> Multi-omics Integration for Discovery of Non-neutral Mutations<a href="multi-omics-integration-for-discovery-of-non-neutral-mutations.html#multi-omics-integration-for-discovery-of-non-neutral-mutations" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Post-Translational Modification (PTM) omics Analysis have identified many PTM sites in pathogens.</p>
<p>Here we demonstrate the application of our package through acetylation proteomics (<span class="citation">Xie et al. (<a href="#ref-xie2015acetylation">2015</a>)</span>) of Mtb conducted by our lab, as well as leveraging recently published O-phosphorylation proteomics data (<span class="citation">Frando et al. (<a href="#ref-frando2023phosphorylation">2023</a>)</span>). We rapidly identify positively selected phosphorylation sites during evolution.</p>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="multi-omics-integration-for-discovery-of-non-neutral-mutations.html#cb44-1" tabindex="-1"></a><span class="co"># The data file must contain at least two required columns: GENE/LOCUS_TAG and AA_POS (amino acid position)</span></span>
<span id="cb44-2"><a href="multi-omics-integration-for-discovery-of-non-neutral-mutations.html#cb44-2" tabindex="-1"></a></span>
<span id="cb44-3"><a href="multi-omics-integration-for-discovery-of-non-neutral-mutations.html#cb44-3" tabindex="-1"></a><span class="co"># Load experimentally validated phosphorylation sites</span></span>
<span id="cb44-4"><a href="multi-omics-integration-for-discovery-of-non-neutral-mutations.html#cb44-4" tabindex="-1"></a>pho_info <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;example/Mtb_pho.csv&quot;</span>)</span>
<span id="cb44-5"><a href="multi-omics-integration-for-discovery-of-non-neutral-mutations.html#cb44-5" tabindex="-1"></a></span>
<span id="cb44-6"><a href="multi-omics-integration-for-discovery-of-non-neutral-mutations.html#cb44-6" tabindex="-1"></a><span class="co"># Load experimentally validated acetylation sites</span></span>
<span id="cb44-7"><a href="multi-omics-integration-for-discovery-of-non-neutral-mutations.html#cb44-7" tabindex="-1"></a>ace_info <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;example/Mtb_ace.csv&quot;</span>)</span>
<span id="cb44-8"><a href="multi-omics-integration-for-discovery-of-non-neutral-mutations.html#cb44-8" tabindex="-1"></a></span>
<span id="cb44-9"><a href="multi-omics-integration-for-discovery-of-non-neutral-mutations.html#cb44-9" tabindex="-1"></a></span>
<span id="cb44-10"><a href="multi-omics-integration-for-discovery-of-non-neutral-mutations.html#cb44-10" tabindex="-1"></a></span>
<span id="cb44-11"><a href="multi-omics-integration-for-discovery-of-non-neutral-mutations.html#cb44-11" tabindex="-1"></a><span class="co"># Using the AddModificationInfo function, we integrate PTM information into the mutation data. This step annotates whether each mutation occurs at known PTM sites.</span></span>
<span id="cb44-12"><a href="multi-omics-integration-for-discovery-of-non-neutral-mutations.html#cb44-12" tabindex="-1"></a>Mtb_mut <span class="ot">&lt;-</span> <span class="fu">AddModificationInfo</span>(Mtb_mut,</span>
<span id="cb44-13"><a href="multi-omics-integration-for-discovery-of-non-neutral-mutations.html#cb44-13" tabindex="-1"></a>                               <span class="at">modification_dt =</span> ace_info,</span>
<span id="cb44-14"><a href="multi-omics-integration-for-discovery-of-non-neutral-mutations.html#cb44-14" tabindex="-1"></a>                               <span class="at">type =</span> <span class="st">&quot;Acetylation&quot;</span>)</span></code></pre></div>
<pre><code>## Add Acetylation info to MutationObject</code></pre>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="multi-omics-integration-for-discovery-of-non-neutral-mutations.html#cb46-1" tabindex="-1"></a>Mtb_mut <span class="ot">&lt;-</span> <span class="fu">AddModificationInfo</span>(Mtb_mut,</span>
<span id="cb46-2"><a href="multi-omics-integration-for-discovery-of-non-neutral-mutations.html#cb46-2" tabindex="-1"></a>                               <span class="at">modification_dt =</span> pho_info,</span>
<span id="cb46-3"><a href="multi-omics-integration-for-discovery-of-non-neutral-mutations.html#cb46-3" tabindex="-1"></a>                               <span class="at">type =</span> <span class="st">&quot;Phosphorylation&quot;</span>)</span></code></pre></div>
<pre><code>## Add Phosphorylation info to MutationObject</code></pre>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb48-1"><a href="multi-omics-integration-for-discovery-of-non-neutral-mutations.html#cb48-1" tabindex="-1"></a><span class="co"># The enhanced mutation annotation now includes two new columns:</span></span>
<span id="cb48-2"><a href="multi-omics-integration-for-discovery-of-non-neutral-mutations.html#cb48-2" tabindex="-1"></a><span class="co"># Acetylation: Indicates if the site is a known acetylation site</span></span>
<span id="cb48-3"><a href="multi-omics-integration-for-discovery-of-non-neutral-mutations.html#cb48-3" tabindex="-1"></a><span class="co"># Phosphorylation: Indicates if the site is a known phosphorylation site</span></span>
<span id="cb48-4"><a href="multi-omics-integration-for-discovery-of-non-neutral-mutations.html#cb48-4" tabindex="-1"></a>Mtb_mut<span class="sc">$</span>snp_list<span class="sc">$</span>snp_anno <span class="sc">%&gt;%</span> <span class="fu">head</span>()</span></code></pre></div>
<pre><code>##             CHROM POS TYPE FTYPE STRAND  NT_POS AA_POS
## A8Csnp  NC_018143   8  snp   CDS      +  8/1524  3/507
## G10Csnp NC_018143  10  snp   CDS      + 10/1524  4/507
## A11Csnp NC_018143  11  snp   CDS      + 11/1524  4/507
## C12Tsnp NC_018143  12  snp   CDS      + 12/1524  4/507
## C15Tsnp NC_018143  15  snp   CDS      + 15/1524  5/507
## G16Csnp NC_018143  16  snp   CDS      + 16/1524  6/507
##                                       EFFECT LOCUS_TAG_NEW GENE
## A8Csnp     missense_variant c.8A&gt;C p.Asp3Ala  RVBD_RS00005 dnaA
## G10Csnp   missense_variant c.10G&gt;C p.Asp4His  RVBD_RS00005 dnaA
## A11Csnp   missense_variant c.11A&gt;C p.Asp4Ala  RVBD_RS00005 dnaA
## C12Tsnp synonymous_variant c.12C&gt;T p.Asp4Asp  RVBD_RS00005 dnaA
## C15Tsnp synonymous_variant c.15C&gt;T p.Pro5Pro  RVBD_RS00005 dnaA
## G16Csnp   missense_variant c.16G&gt;C p.Gly6Arg  RVBD_RS00005 dnaA
##                                                PRODUCT AA_effect_short
## A8Csnp  chromosomal replication initiator protein DnaA             D3A
## G10Csnp chromosomal replication initiator protein DnaA             D4H
## A11Csnp chromosomal replication initiator protein DnaA             D4A
## C12Tsnp chromosomal replication initiator protein DnaA             D4D
## C15Tsnp chromosomal replication initiator protein DnaA             P5P
## G16Csnp chromosomal replication initiator protein DnaA             G6R
##         LOCUS_TAG   INDEX Acetylation Phosphorylation
## A8Csnp     Rv0001  A8Csnp          NA              NA
## G10Csnp    Rv0001 G10Csnp          NA              NA
## A11Csnp    Rv0001 A11Csnp          NA              NA
## C12Tsnp    Rv0001 C12Tsnp          NA              NA
## C15Tsnp    Rv0001 C15Tsnp          NA              NA
## G16Csnp    Rv0001 G16Csnp          NA              NA</code></pre>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb50-1"><a href="multi-omics-integration-for-discovery-of-non-neutral-mutations.html#cb50-1" tabindex="-1"></a><span class="co"># Filter mutations occurring at phosphorylation sites</span></span>
<span id="cb50-2"><a href="multi-omics-integration-for-discovery-of-non-neutral-mutations.html#cb50-2" tabindex="-1"></a>Mtb_mut<span class="sc">$</span>snp_list<span class="sc">$</span>snp_anno <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Phosphorylation<span class="sc">==</span>T) <span class="sc">%&gt;%</span> <span class="fu">head</span>()</span></code></pre></div>
<pre><code>##              CHROM POS TYPE FTYPE STRAND   NT_POS  AA_POS
## A526Gsnp NC_018143 526  snp   CDS      + 526/1524 176/507
## C527Tsnp NC_018143 527  snp   CDS      + 527/1524 176/507
## C629Gsnp NC_018143 629  snp   CDS      + 629/1524 210/507
## C630Tsnp NC_018143 630  snp   CDS      + 630/1524 210/507
## C630Gsnp NC_018143 630  snp   CDS      + 630/1524 210/507
## C876Gsnp NC_018143 876  snp   CDS      + 876/1524 292/507
##                                           EFFECT LOCUS_TAG_NEW GENE
## A526Gsnp   missense_variant c.526A&gt;G p.Thr176Ala  RVBD_RS00005 dnaA
## C527Tsnp   missense_variant c.527C&gt;T p.Thr176Met  RVBD_RS00005 dnaA
## C629Gsnp   missense_variant c.629C&gt;G p.Ser210Cys  RVBD_RS00005 dnaA
## C630Tsnp synonymous_variant c.630C&gt;T p.Ser210Ser  RVBD_RS00005 dnaA
## C630Gsnp synonymous_variant c.630C&gt;G p.Ser210Ser  RVBD_RS00005 dnaA
## C876Gsnp synonymous_variant c.876C&gt;G p.Thr292Thr  RVBD_RS00005 dnaA
##                                                 PRODUCT AA_effect_short
## A526Gsnp chromosomal replication initiator protein DnaA           T176A
## C527Tsnp chromosomal replication initiator protein DnaA           T176M
## C629Gsnp chromosomal replication initiator protein DnaA           S210C
## C630Tsnp chromosomal replication initiator protein DnaA           S210S
## C630Gsnp chromosomal replication initiator protein DnaA           S210S
## C876Gsnp chromosomal replication initiator protein DnaA           T292T
##          LOCUS_TAG    INDEX Acetylation Phosphorylation
## A526Gsnp    Rv0001 A526Gsnp          NA            TRUE
## C527Tsnp    Rv0001 C527Tsnp          NA            TRUE
## C629Gsnp    Rv0001 C629Gsnp          NA            TRUE
## C630Tsnp    Rv0001 C630Tsnp          NA            TRUE
## C630Gsnp    Rv0001 C630Gsnp          NA            TRUE
## C876Gsnp    Rv0001 C876Gsnp          NA            TRUE</code></pre>
<p>The MutNumPlot function enables visualization of mutation frequency distribution for specific genes, with PTM sites highlighted. This visualization helps identify key mutations that may affect protein post-translational modifications.</p>
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb52-1"><a href="multi-omics-integration-for-discovery-of-non-neutral-mutations.html#cb52-1" tabindex="-1"></a><span class="co"># Visualize mutation frequency distributions</span></span>
<span id="cb52-2"><a href="multi-omics-integration-for-discovery-of-non-neutral-mutations.html#cb52-2" tabindex="-1"></a><span class="co"># PTM sites (acetylation and phosphorylation) are highlighted with special colors</span></span>
<span id="cb52-3"><a href="multi-omics-integration-for-discovery-of-non-neutral-mutations.html#cb52-3" tabindex="-1"></a>Mtb_mut<span class="sc">$</span><span class="fu">MutNumPlot</span>(<span class="st">&quot;Rv0001&quot;</span>,<span class="at">Modification =</span> <span class="fu">c</span>(<span class="st">&quot;Acetylation&quot;</span>,<span class="st">&quot;Phosphorylation&quot;</span>))</span></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-13-1.png" width="672" /></p>
<div class="sourceCode" id="cb53"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb53-1"><a href="multi-omics-integration-for-discovery-of-non-neutral-mutations.html#cb53-1" tabindex="-1"></a>Mtb_mut<span class="sc">$</span><span class="fu">MutNumPlot</span>(<span class="st">&quot;Rv0324&quot;</span>,<span class="at">Modification =</span> <span class="fu">c</span>(<span class="st">&quot;Acetylation&quot;</span>,<span class="st">&quot;Phosphorylation&quot;</span>))</span></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-13-2.png" width="672" /></p>
<p>Importantly, we have experimentally demonstrated mutation of a phosphosite in Rv0324, T168A, partially rescued growth arrest during overexpression of <em>Rv0324</em> in <em>Mycobacterium semgmatis</em>, the vanguard of mycobacterial research. This result hightlight the practicality of this simple function.</p>

<div id="refs" class="references csl-bib-body hanging-indent" entry-spacing="0">
<div class="csl-entry">
Frando, Alex, Vatsal Boradia, Marina Gritsenko, Christine Beltejar, Lisa Day, David R Sherman, Si Ma, Jesse M Jacobs, and Christoph Grundner. 2023. <span>“The Mycobacterium Tuberculosis Protein o-Phosphorylation Landscape.”</span> <em>Nature Microbiology</em> 8 (3): 548–61. <a href="https://doi.org/10.1038/s41564-023-01367-z">https://doi.org/10.1038/s41564-023-01367-z</a>.
</div>
<div class="csl-entry">
Liu, Qinqin, Jian Zhu, Colin L Dulberger, Sierra Stanley, Sarah Wilson, Emily S Chung, Xiaojun Wang, et al. 2022. <span>“Tuberculosis Treatment Failure Associated with Evolution of Antibiotic Resilience.”</span> <em>Science</em> 378 (6624): 1111–18. <a href="https://doi.org/10.1126/science.abq2787">https://doi.org/10.1126/science.abq2787</a>.
</div>
<div class="csl-entry">
Xie, Lianhui, Xiaojun Wang, Jia Zeng, Mei Zhou, Xiaohui Duan, Qiaoling Li, Zheng Zhang, et al. 2015. <span>“Proteome-Wide Lysine Acetylation Profiling of the Human Pathogen Mycobacterium Tuberculosis.”</span> <em>International Journal of Biochemistry &amp; Cell Biology</em> 59 (February): 193–202. <a href="https://doi.org/10.1016/j.biocel.2014.11.010">https://doi.org/10.1016/j.biocel.2014.11.010</a>.
</div>
</div>
</div>
<h3>References</h3>
<div id="refs" class="references csl-bib-body hanging-indent" entry-spacing="0">
<div id="ref-frando2023phosphorylation" class="csl-entry">
Frando, Alex, Vatsal Boradia, Marina Gritsenko, Christine Beltejar, Lisa Day, David R Sherman, Si Ma, Jesse M Jacobs, and Christoph Grundner. 2023. <span>“The Mycobacterium Tuberculosis Protein o-Phosphorylation Landscape.”</span> <em>Nature Microbiology</em> 8 (3): 548–61. <a href="https://doi.org/10.1038/s41564-023-01367-z">https://doi.org/10.1038/s41564-023-01367-z</a>.
</div>
<div id="ref-xie2015acetylation" class="csl-entry">
Xie, Lianhui, Xiaojun Wang, Jia Zeng, Mei Zhou, Xiaohui Duan, Qiaoling Li, Zheng Zhang, et al. 2015. <span>“Proteome-Wide Lysine Acetylation Profiling of the Human Pathogen Mycobacterium Tuberculosis.”</span> <em>International Journal of Biochemistry &amp; Cell Biology</em> 59 (February): 193–202. <a href="https://doi.org/10.1016/j.biocel.2014.11.010">https://doi.org/10.1016/j.biocel.2014.11.010</a>.
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="katg-mutation-analysis-isoniazid-resistance.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
  "sharing": {
    "github": false,
    "facebook": true,
    "twitter": true,
    "linkedin": false,
    "weibo": false,
    "instapaper": false,
    "vk": false,
    "whatsapp": false,
    "all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
  },
  "fontsettings": {
    "theme": "white",
    "family": "sans",
    "size": 2
  },
  "edit": {
    "link": "https://github.com/USERNAME/REPO/edit/BRANCH/index.Rmd",
    "text": "Edit"
  },
  "history": {
    "link": null,
    "text": null
  },
  "view": {
    "link": null,
    "text": null
  },
  "download": ["_main.pdf", "_main.epub"],
  "search": {
    "engine": "fuse",
    "options": null
  },
  "toc": {
    "collapse": "subsection"
  }
});
});
</script>

</body>

</html>
