<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 4 Mutation Screening and Analysis | Tutorial for R qMut package (v1.0.0)</title>
  <meta name="description" content="Chapter 4 Mutation Screening and Analysis | Tutorial for R qMut package (v1.0.0)" />
  <meta name="generator" content="bookdown 0.43 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 4 Mutation Screening and Analysis | Tutorial for R qMut package (v1.0.0)" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 4 Mutation Screening and Analysis | Tutorial for R qMut package (v1.0.0)" />
  
  
  

<meta name="author" content="Hongxiang Xu" />


<meta name="date" content="2025-06-20" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="data-integration.html"/>
<link rel="next" href="katg-mutation-analysis-isoniazid-resistance.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>
<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Tutorial for R qMut package</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> About</a></li>
<li class="chapter" data-level="2" data-path="installation.html"><a href="installation.html"><i class="fa fa-check"></i><b>2</b> <strong>Installation</strong></a></li>
<li class="chapter" data-level="3" data-path="data-integration.html"><a href="data-integration.html"><i class="fa fa-check"></i><b>3</b> Data Integration</a>
<ul>
<li class="chapter" data-level="3.1" data-path="data-integration.html"><a href="data-integration.html#input-requirements"><i class="fa fa-check"></i><b>3.1</b> Input Requirements</a></li>
<li class="chapter" data-level="3.2" data-path="data-integration.html"><a href="data-integration.html#creating-the-snp-object"><i class="fa fa-check"></i><b>3.2</b> Creating the SNP Object</a></li>
<li class="chapter" data-level="3.3" data-path="data-integration.html"><a href="data-integration.html#merging-objects-for-expanded-analysis"><i class="fa fa-check"></i><b>3.3</b> Merging Objects for Expanded Analysis</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="mutation-screening-and-analysis.html"><a href="mutation-screening-and-analysis.html"><i class="fa fa-check"></i><b>4</b> Mutation Screening and Analysis</a>
<ul>
<li class="chapter" data-level="4.1" data-path="mutation-screening-and-analysis.html"><a href="mutation-screening-and-analysis.html#basic-mutation-statistics"><i class="fa fa-check"></i><b>4.1</b> Basic Mutation Statistics</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="katg-mutation-analysis-isoniazid-resistance.html"><a href="katg-mutation-analysis-isoniazid-resistance.html"><i class="fa fa-check"></i><b>5</b> katG Mutation Analysis (Isoniazid Resistance)</a></li>
<li class="chapter" data-level="6" data-path="multi-omics-integration-for-discovery-of-non-neutral-mutations.html"><a href="multi-omics-integration-for-discovery-of-non-neutral-mutations.html"><i class="fa fa-check"></i><b>6</b> Multi-omics Integration for Discovery of Non-neutral Mutations</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Tutorial for R qMut package (v1.0.0)</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="mutation-screening-and-analysis" class="section level1 hasAnchor" number="4">
<h1><span class="header-section-number">Chapter 4</span> Mutation Screening and Analysis<a href="mutation-screening-and-analysis.html#mutation-screening-and-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>The <code>Mtb_mut</code> object serves as the primary data structure for all downstream analyses in our package. We provide pre-processed qs objects for several common bacterial species:</p>
<div id="basic-mutation-statistics" class="section level2 hasAnchor" number="4.1">
<h2><span class="header-section-number">4.1</span> Basic Mutation Statistics<a href="mutation-screening-and-analysis.html#basic-mutation-statistics" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="mutation-screening-and-analysis.html#cb16-1" tabindex="-1"></a><span class="co"># Calculate occurrence frequency for all mutations</span></span>
<span id="cb16-2"><a href="mutation-screening-and-analysis.html#cb16-2" tabindex="-1"></a>all_mut <span class="ot">&lt;-</span> Mtb_mut<span class="sc">$</span><span class="fu">estimate_imp</span>(<span class="st">&quot;all&quot;</span>)</span>
<span id="cb16-3"><a href="mutation-screening-and-analysis.html#cb16-3" tabindex="-1"></a></span>
<span id="cb16-4"><a href="mutation-screening-and-analysis.html#cb16-4" tabindex="-1"></a><span class="fu">head</span>(all_mut)</span></code></pre></div>
<pre><code>##             CHROM POS TYPE FTYPE STRAND  NT_POS AA_POS
## A8Csnp  NC_018143   8  snp   CDS      +  8/1524  3/507
## G10Csnp NC_018143  10  snp   CDS      + 10/1524  4/507
## A11Csnp NC_018143  11  snp   CDS      + 11/1524  4/507
## C12Tsnp NC_018143  12  snp   CDS      + 12/1524  4/507
## C15Tsnp NC_018143  15  snp   CDS      + 15/1524  5/507
## G16Csnp NC_018143  16  snp   CDS      + 16/1524  6/507
##                                       EFFECT LOCUS_TAG_NEW GENE
## A8Csnp     missense_variant c.8A&gt;C p.Asp3Ala  RVBD_RS00005 dnaA
## G10Csnp   missense_variant c.10G&gt;C p.Asp4His  RVBD_RS00005 dnaA
## A11Csnp   missense_variant c.11A&gt;C p.Asp4Ala  RVBD_RS00005 dnaA
## C12Tsnp synonymous_variant c.12C&gt;T p.Asp4Asp  RVBD_RS00005 dnaA
## C15Tsnp synonymous_variant c.15C&gt;T p.Pro5Pro  RVBD_RS00005 dnaA
## G16Csnp   missense_variant c.16G&gt;C p.Gly6Arg  RVBD_RS00005 dnaA
##                                                PRODUCT AA_effect_short
## A8Csnp  chromosomal replication initiator protein DnaA             D3A
## G10Csnp chromosomal replication initiator protein DnaA             D4H
## A11Csnp chromosomal replication initiator protein DnaA             D4A
## C12Tsnp chromosomal replication initiator protein DnaA             D4D
## C15Tsnp chromosomal replication initiator protein DnaA             P5P
## G16Csnp chromosomal replication initiator protein DnaA             G6R
##         LOCUS_TAG   INDEX Times
## A8Csnp     Rv0001  A8Csnp     2
## G10Csnp    Rv0001 G10Csnp     1
## A11Csnp    Rv0001 A11Csnp    28
## C12Tsnp    Rv0001 C12Tsnp     1
## C15Tsnp    Rv0001 C15Tsnp     2
## G16Csnp    Rv0001 G16Csnp     1</code></pre>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="mutation-screening-and-analysis.html#cb18-1" tabindex="-1"></a><span class="co"># Analyze mutations in Rv0324 (default: coding sequence only)</span></span>
<span id="cb18-2"><a href="mutation-screening-and-analysis.html#cb18-2" tabindex="-1"></a>Rv0324_mut <span class="ot">&lt;-</span> Mtb_mut<span class="sc">$</span><span class="fu">estimate_imp</span>(<span class="st">&quot;Rv0324&quot;</span>, <span class="at">gene_type =</span> <span class="st">&quot;LOCUS_TAG&quot;</span>)</span></code></pre></div>
<pre><code>## filter genes</code></pre>
<pre><code>## remain 51181 sample and 191 snps</code></pre>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="mutation-screening-and-analysis.html#cb21-1" tabindex="-1"></a><span class="fu">head</span>(Rv0324_mut)</span></code></pre></div>
<pre><code>##                 CHROM    POS TYPE FTYPE STRAND NT_POS AA_POS
## A391352Tsnp NC_018143 391352  snp   CDS      +  1/681  1/226
## G391354Csnp NC_018143 391354  snp   CDS      +  3/681  1/226
## G391355Asnp NC_018143 391355  snp   CDS      +  4/681  2/226
## C391356Asnp NC_018143 391356  snp   CDS      +  5/681  2/226
## G391359Asnp NC_018143 391359  snp   CDS      +  8/681  3/226
## C391365Tsnp NC_018143 391365  snp   CDS      + 14/681  5/226
##                                             EFFECT LOCUS_TAG_NEW   GENE
## A391352Tsnp initiator_codon_variant c.1A&gt;T p.Met1?  RVBD_RS01710 Rv0324
## G391354Csnp initiator_codon_variant c.3G&gt;C p.Met1?  RVBD_RS01710 Rv0324
## G391355Asnp      missense_variant c.4G&gt;A p.Ala2Thr  RVBD_RS01710 Rv0324
## C391356Asnp      missense_variant c.5C&gt;A p.Ala2Asp  RVBD_RS01710 Rv0324
## G391359Asnp      missense_variant c.8G&gt;A p.Gly3Glu  RVBD_RS01710 Rv0324
## C391365Tsnp     missense_variant c.14C&gt;T p.Ser5Phe  RVBD_RS01710 Rv0324
##                                                            PRODUCT
## A391352Tsnp metalloregulator ArsR/SmtB family transcription factor
## G391354Csnp metalloregulator ArsR/SmtB family transcription factor
## G391355Asnp metalloregulator ArsR/SmtB family transcription factor
## C391356Asnp metalloregulator ArsR/SmtB family transcription factor
## G391359Asnp metalloregulator ArsR/SmtB family transcription factor
## C391365Tsnp metalloregulator ArsR/SmtB family transcription factor
##             AA_effect_short LOCUS_TAG       INDEX Times
## A391352Tsnp             M1?    Rv0324 A391352Tsnp     1
## G391354Csnp             M1?    Rv0324 G391354Csnp     4
## G391355Asnp             A2T    Rv0324 G391355Asnp     1
## C391356Asnp             A2D    Rv0324 C391356Asnp     1
## G391359Asnp             G3E    Rv0324 G391359Asnp     1
## C391365Tsnp             S5F    Rv0324 C391365Tsnp     1</code></pre>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="mutation-screening-and-analysis.html#cb23-1" tabindex="-1"></a><span class="co"># Include 300bp upstream regulatory region</span></span>
<span id="cb23-2"><a href="mutation-screening-and-analysis.html#cb23-2" tabindex="-1"></a>Rv0324_mut <span class="ot">&lt;-</span> Mtb_mut<span class="sc">$</span><span class="fu">estimate_imp</span>(<span class="at">gene =</span> <span class="st">&quot;Rv0324&quot;</span>,</span>
<span id="cb23-3"><a href="mutation-screening-and-analysis.html#cb23-3" tabindex="-1"></a>                                   <span class="at">gene_type =</span> <span class="st">&quot;LOCUS_TAG&quot;</span>,</span>
<span id="cb23-4"><a href="mutation-screening-and-analysis.html#cb23-4" tabindex="-1"></a>                                   <span class="at">upstream_length =</span> <span class="dv">300</span>)</span></code></pre></div>
<pre><code>## filter genes</code></pre>
<pre><code>## remain 51181 sample and 293 snps</code></pre>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="data-integration.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="katg-mutation-analysis-isoniazid-resistance.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
  "sharing": {
    "github": false,
    "facebook": true,
    "twitter": true,
    "linkedin": false,
    "weibo": false,
    "instapaper": false,
    "vk": false,
    "whatsapp": false,
    "all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
  },
  "fontsettings": {
    "theme": "white",
    "family": "sans",
    "size": 2
  },
  "edit": {
    "link": "https://github.com/USERNAME/REPO/edit/BRANCH/index.Rmd",
    "text": "Edit"
  },
  "history": {
    "link": null,
    "text": null
  },
  "view": {
    "link": null,
    "text": null
  },
  "download": ["_main.pdf", "_main.epub"],
  "search": {
    "engine": "fuse",
    "options": null
  },
  "toc": {
    "collapse": "subsection"
  }
});
});
</script>

</body>

</html>
