<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 3 Data Integration | Tutorial for R qMut package (v1.0.0)</title>
  <meta name="description" content="Chapter 3 Data Integration | Tutorial for R qMut package (v1.0.0)" />
  <meta name="generator" content="bookdown 0.43 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 3 Data Integration | Tutorial for R qMut package (v1.0.0)" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 3 Data Integration | Tutorial for R qMut package (v1.0.0)" />
  
  
  

<meta name="author" content="Hongxiang Xu" />


<meta name="date" content="2025-06-20" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="installation.html"/>
<link rel="next" href="mutation-screening-and-analysis.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>
<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Tutorial for R qMut package</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> About</a></li>
<li class="chapter" data-level="2" data-path="installation.html"><a href="installation.html"><i class="fa fa-check"></i><b>2</b> <strong>Installation</strong></a></li>
<li class="chapter" data-level="3" data-path="data-integration.html"><a href="data-integration.html"><i class="fa fa-check"></i><b>3</b> Data Integration</a>
<ul>
<li class="chapter" data-level="3.1" data-path="data-integration.html"><a href="data-integration.html#input-requirements"><i class="fa fa-check"></i><b>3.1</b> Input Requirements</a></li>
<li class="chapter" data-level="3.2" data-path="data-integration.html"><a href="data-integration.html#creating-the-snp-object"><i class="fa fa-check"></i><b>3.2</b> Creating the SNP Object</a></li>
<li class="chapter" data-level="3.3" data-path="data-integration.html"><a href="data-integration.html#merging-objects-for-expanded-analysis"><i class="fa fa-check"></i><b>3.3</b> Merging Objects for Expanded Analysis</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="mutation-screening-and-analysis.html"><a href="mutation-screening-and-analysis.html"><i class="fa fa-check"></i><b>4</b> Mutation Screening and Analysis</a>
<ul>
<li class="chapter" data-level="4.1" data-path="mutation-screening-and-analysis.html"><a href="mutation-screening-and-analysis.html#basic-mutation-statistics"><i class="fa fa-check"></i><b>4.1</b> Basic Mutation Statistics</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="katg-mutation-analysis-isoniazid-resistance.html"><a href="katg-mutation-analysis-isoniazid-resistance.html"><i class="fa fa-check"></i><b>5</b> katG Mutation Analysis (Isoniazid Resistance)</a></li>
<li class="chapter" data-level="6" data-path="multi-omics-integration-for-discovery-of-non-neutral-mutations.html"><a href="multi-omics-integration-for-discovery-of-non-neutral-mutations.html"><i class="fa fa-check"></i><b>6</b> Multi-omics Integration for Discovery of Non-neutral Mutations</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Tutorial for R qMut package (v1.0.0)</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="data-integration" class="section level1 hasAnchor" number="3">
<h1><span class="header-section-number">Chapter 3</span> Data Integration<a href="data-integration.html#data-integration" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="input-requirements" class="section level2 hasAnchor" number="3.1">
<h2><span class="header-section-number">3.1</span> Input Requirements<a href="data-integration.html#input-requirements" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The package accepts mutation data in either CSV (from snippy) or standard VCF format. A minimal input should contain these columns:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="data-integration.html#cb2-1" tabindex="-1"></a><span class="fu">library</span>(qs)</span>
<span id="cb2-2"><a href="data-integration.html#cb2-2" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb2-3"><a href="data-integration.html#cb2-3" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span></code></pre></div>
<pre><code>## ── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
## ✔ forcats   1.0.0     ✔ stringr   1.5.1
## ✔ lubridate 1.9.3     ✔ tidyr     1.3.1
## ✔ readr     2.1.5     
## ── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
## ✖ dplyr::between()     masks data.table::between()
## ✖ tidyr::expand()      masks ggtree::expand(), Matrix::expand()
## ✖ tidyr::extract()     masks magrittr::extract()
## ✖ dplyr::filter()      masks stats::filter()
## ✖ dplyr::first()       masks data.table::first()
## ✖ lubridate::hour()    masks data.table::hour()
## ✖ ggtree::inset()      masks magrittr::inset()
## ✖ lubridate::isoweek() masks data.table::isoweek()
## ✖ dplyr::lag()         masks stats::lag()
## ✖ dplyr::last()        masks data.table::last()
## ✖ lubridate::mday()    masks data.table::mday()
## ✖ lubridate::minute()  masks data.table::minute()
## ✖ lubridate::month()   masks data.table::month()
## ✖ tidyr::pack()        masks Matrix::pack()
## ✖ lubridate::quarter() masks data.table::quarter()
## ✖ lubridate::second()  masks data.table::second()
## ✖ purrr::set_names()   masks magrittr::set_names()
## ✖ lubridate::stamp()   masks cowplot::stamp()
## ✖ purrr::transpose()   masks data.table::transpose()
## ✖ tidyr::unpack()      masks Matrix::unpack()
## ✖ lubridate::wday()    masks data.table::wday()
## ✖ lubridate::week()    masks data.table::week()
## ✖ ape::where()         masks dplyr::where()
## ✖ lubridate::yday()    masks data.table::yday()
## ✖ lubridate::year()    masks data.table::year()
## ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors</code></pre>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="data-integration.html#cb4-1" tabindex="-1"></a>exp_data <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;example/snippy_data/snps.csv&quot;</span>)</span>
<span id="cb4-2"><a href="data-integration.html#cb4-2" tabindex="-1"></a></span>
<span id="cb4-3"><a href="data-integration.html#cb4-3" tabindex="-1"></a><span class="fu">head</span>(exp_data,<span class="at">n=</span><span class="dv">3</span>)</span></code></pre></div>
<pre><code>##       CHROM  POS TYPE REF ALT EVIDENCE FTYPE STRAND   NT_POS  AA_POS
## 1 NC_000962 1718  snp   C   A A:20 C:0                              
## 2 NC_000962 1977  snp   A   G G:20 A:0                              
## 3 NC_000962 2532  snp   T   C C:20 T:0   CDS      + 481/1209 161/402
##                                    EFFECT LOCUS_TAG GENE
## 1                                                       
## 2                                                       
## 3 synonymous_variant c.481T&gt;C p.Leu161Leu    Rv0002 dnaN
##                           PRODUCT
## 1                                
## 2                                
## 3 DNA polymerase III subunit beta</code></pre>
<p>The <code>snp_file_list</code> contains mutation information identified by the snippy software. A standard mutation file should include the following 14 columns:</p>
<table>
<thead>
<tr class="header">
<th>Column</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>CHROM</code></td>
<td>Chromosome identifier</td>
</tr>
<tr class="even">
<td><code>POS</code></td>
<td>Genomic position of the mutation</td>
</tr>
<tr class="odd">
<td><code>TYPE</code></td>
<td>Mutation type (SNP, INDEL, etc.)</td>
</tr>
<tr class="even">
<td><code>REF</code></td>
<td>Reference allele</td>
</tr>
<tr class="odd">
<td><code>ALT</code></td>
<td>Alternate allele</td>
</tr>
<tr class="even">
<td><code>EVIDENCE</code></td>
<td>Read depth supporting the mutation</td>
</tr>
<tr class="odd">
<td><code>FTYPE</code></td>
<td>Functional type of mutation</td>
</tr>
<tr class="even">
<td><code>STRAND</code></td>
<td>Coding strand (+/-)</td>
</tr>
<tr class="odd">
<td><code>NT_POS</code></td>
<td>Nucleotide position within gene</td>
</tr>
<tr class="even">
<td><code>AA_POS</code></td>
<td>Amino acid position in protein</td>
</tr>
<tr class="odd">
<td><code>EFFECT</code></td>
<td>Functional consequence</td>
</tr>
<tr class="even">
<td><code>LOCUS_TAG</code></td>
<td>Gene locus identifier</td>
</tr>
<tr class="odd">
<td><code>GENE</code></td>
<td>Gene name</td>
</tr>
<tr class="even">
<td><code>PRODUCT</code></td>
<td>Protein product description</td>
</tr>
</tbody>
</table>
<p>Mutation matrices are typically represented as binary (0/1) indicators denoting presence/absence of mutations across samples. However, three key observations motivate our storage approach:</p>
<ol style="list-style-type: decimal">
<li><strong>Sparsity</strong>: Most mutations occur in only a few samples</li>
<li><strong>Annotation redundancy</strong>: Many mutation types are identical across samples</li>
<li><strong>Scalability</strong>: Whole-genome datasets can be extremely large</li>
</ol>
<p>We therefore decompose the data into two components:</p>
<ol style="list-style-type: decimal">
<li><strong><code>snp_anno</code></strong>: Stores the union of all mutation annotations (shared across samples)</li>
<li><strong><code>snp_data</code></strong>: Stores the sparse binary matrix of mutation occurrences</li>
</ol>
<p>This decomposition provides significant advantages:</p>
<ul>
<li>Reduced storage requirements (only unique annotations stored once)</li>
<li>Faster matrix operations on sparse data</li>
<li>Efficient memory usage during analysis</li>
</ul>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="data-integration.html#cb6-1" tabindex="-1"></a>all_smp_name <span class="ot">&lt;-</span> <span class="fu">str_extract</span>(snp_file_list,<span class="st">&quot;(?&lt;=snippy_data/)[^/]+&quot;</span>)</span>
<span id="cb6-2"><a href="data-integration.html#cb6-2" tabindex="-1"></a></span>
<span id="cb6-3"><a href="data-integration.html#cb6-3" tabindex="-1"></a><span class="fu">IntegrateData</span>(</span>
<span id="cb6-4"><a href="data-integration.html#cb6-4" tabindex="-1"></a>  <span class="at">file_path =</span> snp_file_list,</span>
<span id="cb6-5"><a href="data-integration.html#cb6-5" tabindex="-1"></a>  <span class="at">smp_name =</span> sample_names,</span>
<span id="cb6-6"><a href="data-integration.html#cb6-6" tabindex="-1"></a>  <span class="at">output_dir =</span> <span class="st">&quot;output&quot;</span>,</span>
<span id="cb6-7"><a href="data-integration.html#cb6-7" tabindex="-1"></a>  <span class="at">chunk_size =</span> <span class="dv">1000</span>,  <span class="co"># Process in chunks of 1000 samples to reduce memory usage</span></span>
<span id="cb6-8"><a href="data-integration.html#cb6-8" tabindex="-1"></a>  <span class="at">nworkers =</span> <span class="dv">4</span>        <span class="co"># Parallel processing</span></span>
<span id="cb6-9"><a href="data-integration.html#cb6-9" tabindex="-1"></a>)</span></code></pre></div>
<p>The <strong>IntegrateData</strong> function generates two files:snp_anno and snp_data, as mentioned above.</p>
<p>For large-scale analyses (&gt;10,000 samples), we recommend processing in chunks (set chunk_size appropriately).</p>
<p>This approach has been tested with &gt;50,000 bacterial genomes while maintaining memory efficiency.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="data-integration.html#cb7-1" tabindex="-1"></a>snp_anno <span class="ot">&lt;-</span> <span class="fu">qread</span>(<span class="st">&quot;example/snp_anno.qs&quot;</span>,<span class="at">nthreads =</span> <span class="dv">8</span>)</span>
<span id="cb7-2"><a href="data-integration.html#cb7-2" tabindex="-1"></a></span>
<span id="cb7-3"><a href="data-integration.html#cb7-3" tabindex="-1"></a><span class="co"># edit Mtb gene names</span></span>
<span id="cb7-4"><a href="data-integration.html#cb7-4" tabindex="-1"></a>snp_anno<span class="sc">$</span>LOCUS_TAG <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;VBD_&quot;</span>,<span class="st">&quot;v&quot;</span>,snp_anno<span class="sc">$</span>LOCUS_TAG)</span>
<span id="cb7-5"><a href="data-integration.html#cb7-5" tabindex="-1"></a>snp_anno<span class="sc">$</span>GENE <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">str_detect</span>(snp_anno<span class="sc">$</span>GENE,<span class="st">&quot;RVBD_RS&quot;</span>),snp_anno<span class="sc">$</span>LOCUS_TAG,snp_anno<span class="sc">$</span>GENE)</span>
<span id="cb7-6"><a href="data-integration.html#cb7-6" tabindex="-1"></a></span>
<span id="cb7-7"><a href="data-integration.html#cb7-7" tabindex="-1"></a><span class="fu">head</span>(snp_anno)</span></code></pre></div>
<pre><code>##             CHROM POS TYPE FTYPE STRAND  NT_POS AA_POS
## A8Csnp  NC_018143   8  snp   CDS      +  8/1524  3/507
## G10Csnp NC_018143  10  snp   CDS      + 10/1524  4/507
## A11Csnp NC_018143  11  snp   CDS      + 11/1524  4/507
## C12Tsnp NC_018143  12  snp   CDS      + 12/1524  4/507
## C15Tsnp NC_018143  15  snp   CDS      + 15/1524  5/507
## G16Csnp NC_018143  16  snp   CDS      + 16/1524  6/507
##                                       EFFECT LOCUS_TAG_NEW GENE
## A8Csnp     missense_variant c.8A&gt;C p.Asp3Ala  RVBD_RS00005 dnaA
## G10Csnp   missense_variant c.10G&gt;C p.Asp4His  RVBD_RS00005 dnaA
## A11Csnp   missense_variant c.11A&gt;C p.Asp4Ala  RVBD_RS00005 dnaA
## C12Tsnp synonymous_variant c.12C&gt;T p.Asp4Asp  RVBD_RS00005 dnaA
## C15Tsnp synonymous_variant c.15C&gt;T p.Pro5Pro  RVBD_RS00005 dnaA
## G16Csnp   missense_variant c.16G&gt;C p.Gly6Arg  RVBD_RS00005 dnaA
##                                                PRODUCT AA_effect_short
## A8Csnp  chromosomal replication initiator protein DnaA             D3A
## G10Csnp chromosomal replication initiator protein DnaA             D4H
## A11Csnp chromosomal replication initiator protein DnaA             D4A
## C12Tsnp chromosomal replication initiator protein DnaA             D4D
## C15Tsnp chromosomal replication initiator protein DnaA             P5P
## G16Csnp chromosomal replication initiator protein DnaA             G6R
##         LOCUS_TAG
## A8Csnp     Rv0001
## G10Csnp    Rv0001
## A11Csnp    Rv0001
## C12Tsnp    Rv0001
## C15Tsnp    Rv0001
## G16Csnp    Rv0001</code></pre>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="data-integration.html#cb9-1" tabindex="-1"></a>snp_data <span class="ot">&lt;-</span> <span class="fu">qread</span>(<span class="st">&quot;example/snp_data.qs&quot;</span>,<span class="at">nthreads =</span> <span class="dv">8</span>)</span>
<span id="cb9-2"><a href="data-integration.html#cb9-2" tabindex="-1"></a></span>
<span id="cb9-3"><a href="data-integration.html#cb9-3" tabindex="-1"></a>snp_data[<span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>]</span></code></pre></div>
<pre><code>## 6 x 6 sparse Matrix of class &quot;dgCMatrix&quot;
##           A8Csnp G10Csnp A11Csnp C12Tsnp C15Tsnp G16Csnp
## DRR034340      .       .       .       .       .       .
## DRR034341      .       .       .       .       .       .
## DRR034342      .       .       .       .       1       .
## DRR034343      .       .       .       .       .       .
## DRR034344      .       .       .       .       .       .
## DRR034345      .       .       .       .       .       .</code></pre>
</div>
<div id="creating-the-snp-object" class="section level2 hasAnchor" number="3.2">
<h2><span class="header-section-number">3.2</span> Creating the SNP Object<a href="data-integration.html#creating-the-snp-object" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="data-integration.html#cb11-1" tabindex="-1"></a>sample_info <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">&quot;example/sample_info.txt&quot;</span>,<span class="at">data.table =</span> F) <span class="sc">%&gt;%</span></span>
<span id="cb11-2"><a href="data-integration.html#cb11-2" tabindex="-1"></a>  <span class="fu">filter</span>(ID <span class="sc">%in%</span> <span class="fu">rownames</span>(snp_data)) <span class="sc">%&gt;%</span></span>
<span id="cb11-3"><a href="data-integration.html#cb11-3" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">Sample_name=</span>ID) <span class="sc">%&gt;%</span> </span>
<span id="cb11-4"><a href="data-integration.html#cb11-4" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Lineage=</span><span class="fu">paste0</span>(<span class="st">&quot;L&quot;</span>,<span class="fu">str_extract</span>(Lineage,<span class="st">&quot;(?&lt;=LINEAGE).+&quot;</span>)))</span>
<span id="cb11-5"><a href="data-integration.html#cb11-5" tabindex="-1"></a></span>
<span id="cb11-6"><a href="data-integration.html#cb11-6" tabindex="-1"></a><span class="fu">head</span>(sample_info)</span></code></pre></div>
<pre><code>##   Sample_name Genome_coverage Sequencing_depth Lineage     Sublineage BioPrject
## 1   DRR034340        0.976338          52.9203      L2   lineage2.2.1 PRJDB3875
## 2   DRR034341        0.971547          36.7679      L2   lineage2.2.1 PRJDB3875
## 3   DRR034342        0.981057          40.2434      L4 lineage4.9.x.x PRJDB3875
## 4   DRR034343        0.977891          62.5146      L2   lineage2.2.1 PRJDB3875
## 5   DRR034344        0.982537          48.4552      L4 lineage4.9.x.x PRJDB3875
## 6   DRR034345        0.976601          42.5142      L4     lineage4.5 PRJDB3875</code></pre>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="data-integration.html#cb13-1" tabindex="-1"></a>Mtb_mut <span class="ot">&lt;-</span> <span class="fu">CreateSNPObj</span>(<span class="at">snp_anno =</span> snp_anno,</span>
<span id="cb13-2"><a href="data-integration.html#cb13-2" tabindex="-1"></a>                        <span class="at">snp_data =</span> snp_data,</span>
<span id="cb13-3"><a href="data-integration.html#cb13-3" tabindex="-1"></a>                        <span class="at">metadata =</span> sample_info,</span>
<span id="cb13-4"><a href="data-integration.html#cb13-4" tabindex="-1"></a>                        <span class="at">remove_genes_dt =</span> <span class="st">&quot;Mtb&quot;</span>,</span>
<span id="cb13-5"><a href="data-integration.html#cb13-5" tabindex="-1"></a>                        <span class="at">verbose=</span>T)</span></code></pre></div>
<pre><code>## [1/2] Remove selected genes</code></pre>
<pre><code>## [2/2] Annotate amino acid changes</code></pre>
<p>The qMut package utilizes R6 objects that contain multiple accessible attributes and methods for comprehensive mutation analysis.</p>
</div>
<div id="merging-objects-for-expanded-analysis" class="section level2 hasAnchor" number="3.3">
<h2><span class="header-section-number">3.3</span> Merging Objects for Expanded Analysis<a href="data-integration.html#merging-objects-for-expanded-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>When incorporating new samples, the MergeData() function enables seamless integration of multiple qMut objects.</p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="installation.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="mutation-screening-and-analysis.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
  "sharing": {
    "github": false,
    "facebook": true,
    "twitter": true,
    "linkedin": false,
    "weibo": false,
    "instapaper": false,
    "vk": false,
    "whatsapp": false,
    "all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
  },
  "fontsettings": {
    "theme": "white",
    "family": "sans",
    "size": 2
  },
  "edit": {
    "link": "https://github.com/USERNAME/REPO/edit/BRANCH/index.Rmd",
    "text": "Edit"
  },
  "history": {
    "link": null,
    "text": null
  },
  "view": {
    "link": null,
    "text": null
  },
  "download": ["_main.pdf", "_main.epub"],
  "search": {
    "engine": "fuse",
    "options": null
  },
  "toc": {
    "collapse": "subsection"
  }
});
});
</script>

</body>

</html>
