<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="The Mutation object is created to apply all functions for mutation analysis"><title>The Mutation object for snp_data and snp_anno — SNP • qMut</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="The Mutation object for snp_data and snp_anno — SNP"><meta property="og:description" content="The Mutation object is created to apply all functions for mutation analysis"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">qMut</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/my-vignette.html">my-vignette</a>
  </div>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/HongxiangXu/qMut/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>The Mutation object for snp_data and snp_anno</h1>
      <small class="dont-index">Source: <a href="https://github.com/HongxiangXu/qMut/blob/HEAD/R/Mutation_object.R" class="external-link"><code>R/Mutation_object.R</code></a></small>
      <div class="d-none name"><code>SNP.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>The Mutation object is created to apply all functions for mutation analysis</p>
    </div>


    <div class="section level2">
    <h2 id="public-fields">Public fields<a class="anchor" aria-label="anchor" href="#public-fields"></a></h2>
    <p></p><div class="r6-fields"><dl><dt><code>snp_list</code></dt>
<dd><p>A numeric field</p></dd>


<dt><code>metadata</code></dt>
<dd><p>A numeric field</p></dd>


<dt><code>modification_info</code></dt>
<dd><p>A non-numeric field</p></dd>


<dt><code>nj_distance</code></dt>
<dd><p>NJ distance
Initialize a new MyClass object</p></dd>


</dl><p></p></div>
    </div>
    <div class="section level2">
    <h2 id="methods">Methods<a class="anchor" aria-label="anchor" href="#methods"></a></h2>
    
<div class="section">
<h3 id="public-methods">Public methods<a class="anchor" aria-label="anchor" href="#public-methods"></a></h3>

<ul><li><p><a href="#method-Mutation%20Object-new"><code>SNP$new()</code></a></p></li>
<li><p><a href="#method-Mutation%20Object-get_metadata"><code>SNP$get_metadata()</code></a></p></li>
<li><p><a href="#method-Mutation%20Object-ML_data_dt"><code>SNP$ML_data_dt()</code></a></p></li>
<li><p><a href="#method-Mutation%20Object-Evolution_dt"><code>SNP$Evolution_dt()</code></a></p></li>
<li><p><a href="#method-Mutation%20Object-filter_snp"><code>SNP$filter_snp()</code></a></p></li>
<li><p><a href="#method-Mutation%20Object-filter_sample"><code>SNP$filter_sample()</code></a></p></li>
<li><p><a href="#method-Mutation%20Object-estimate_imp_all"><code>SNP$estimate_imp_all()</code></a></p></li>
<li><p><a href="#method-Mutation%20Object-estimate_imp"><code>SNP$estimate_imp()</code></a></p></li>
<li><p><a href="#method-Mutation%20Object-MutTree"><code>SNP$MutTree()</code></a></p></li>
<li><p><a href="#method-Mutation%20Object-MutNumPlot"><code>SNP$MutNumPlot()</code></a></p></li>
<li><p><a href="#method-Mutation%20Object-calculate_NJ"><code>SNP$calculate_NJ()</code></a></p></li>
<li><p><a href="#method-Mutation%20Object-clone"><code>SNP$clone()</code></a></p></li>
</ul></div><p></p><hr><a id="method-Mutation Object-new"></a><div class="section">
<h3 id="method-new-">Method <code>new()</code><a class="anchor" aria-label="anchor" href="#method-new-"></a></h3>

<div class="section">
<h4 id="usage">Usage<a class="anchor" aria-label="anchor" href="#usage"></a></h4>
<p></p><div class="r"><div class="sourceCode"><pre><code><span><span class="va"><a href="../reference/SNP.html">SNP</a></span><span class="op">$</span><span class="fu">new</span><span class="op">(</span></span>
<span>  <span class="va">snp_anno</span>,</span>
<span>  <span class="va">snp_data</span>,</span>
<span>  metadata <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  remove_genes_dt <span class="op">=</span> <span class="cn">NA</span>,</span>
<span>  annotate_AA <span class="op">=</span> <span class="cn">T</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">F</span></span>
<span><span class="op">)</span></span></code></pre></div><p></p></div>
</div>

<div class="section">
<h4 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h4>
<p></p><div class="arguments"><dl><dt><code>snp_anno</code></dt>
<dd><p>Initial value for the field snp_anno</p></dd>


<dt><code>snp_data</code></dt>
<dd><p>Initial value for the field snp_data</p></dd>


<dt><code>metadata</code></dt>
<dd><p>Initial value for the field metadata</p></dd>


<dt><code>remove_genes_dt</code></dt>
<dd><p>Initial value for the field remove_genes_dt</p></dd>


<dt><code>annotate_AA</code></dt>
<dd><p>Initial value for the field remove_genes_dt</p></dd>


<dt><code>verbose</code></dt>
<dd><p>Initial value for the field remove_genes_dt
get_metadata</p></dd>


<dt><code>modification_info</code></dt>
<dd><p>Initial value for the field remove_genes_dt</p></dd>


</dl><p></p></div>
</div>

</div><p></p><hr><a id="method-Mutation Object-get_metadata"></a><div class="section">
<h3 id="method-get-metadata-">Method <code>get_metadata()</code><a class="anchor" aria-label="anchor" href="#method-get-metadata-"></a></h3>

<div class="section">
<h4 id="usage-1">Usage<a class="anchor" aria-label="anchor" href="#usage-1"></a></h4>
<p></p><div class="r"><div class="sourceCode"><pre><code><span><span class="va">SNP</span><span class="op">$</span><span class="fu">get_metadata</span><span class="op">(</span><span class="op">)</span></span></code></pre></div><p></p></div>
</div>

<div class="section">
<h4 id="arguments-1">Arguments<a class="anchor" aria-label="anchor" href="#arguments-1"></a></h4>
<p></p><div class="arguments"><dl><dt><code>Initialize</code></dt>
<dd><p>a new MyClass object</p></dd>


</dl><p></p></div>
</div>

</div><p></p><hr><a id="method-Mutation Object-ML_data_dt"></a><div class="section">
<h3 id="method-ml-data-dt-">Method <code>ML_data_dt()</code><a class="anchor" aria-label="anchor" href="#method-ml-data-dt-"></a></h3>

<div class="section">
<h4 id="usage-2">Usage<a class="anchor" aria-label="anchor" href="#usage-2"></a></h4>
<p></p><div class="r"><div class="sourceCode"><pre><code><span><span class="va">SNP</span><span class="op">$</span><span class="fu">ML_data_dt</span><span class="op">(</span><span class="va">selected_pheno</span><span class="op">)</span></span></code></pre></div><p></p></div>
</div>

<div class="section">
<h4 id="arguments-2">Arguments<a class="anchor" aria-label="anchor" href="#arguments-2"></a></h4>
<p></p><div class="arguments"><dl><dt><code>selected_pheno</code></dt>
<dd><p>Initial value for the field selected_pheno</p></dd>


</dl><p></p></div>
</div>
<div class="section">
<h4 id="returns">Returns<a class="anchor" aria-label="anchor" href="#returns"></a></h4>
<p>NULL
Initialize a new MyClass object</p>
</div>

</div><p></p><hr><a id="method-Mutation Object-Evolution_dt"></a><div class="section">
<h3 id="method-evolution-dt-">Method <code>Evolution_dt()</code><a class="anchor" aria-label="anchor" href="#method-evolution-dt-"></a></h3>

<div class="section">
<h4 id="usage-3">Usage<a class="anchor" aria-label="anchor" href="#usage-3"></a></h4>
<p></p><div class="r"><div class="sourceCode"><pre><code><span><span class="va">SNP</span><span class="op">$</span><span class="fu">Evolution_dt</span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span></code></pre></div><p></p></div>
</div>

<div class="section">
<h4 id="arguments-3">Arguments<a class="anchor" aria-label="anchor" href="#arguments-3"></a></h4>
<p></p><div class="arguments"><dl><dt><code>x</code></dt>
<dd><p>Initial value for the field x
Getting SNPs in certain Genes, or certain SNP in all samples</p></dd>


</dl><p></p></div>
</div>

</div><p></p><hr><a id="method-Mutation Object-filter_snp"></a><div class="section">
<h3 id="method-filter-snp-">Method <code>filter_snp()</code><a class="anchor" aria-label="anchor" href="#method-filter-snp-"></a></h3>

<div class="section">
<h4 id="usage-4">Usage<a class="anchor" aria-label="anchor" href="#usage-4"></a></h4>
<p></p><div class="r"><div class="sourceCode"><pre><code><span><span class="va">SNP</span><span class="op">$</span><span class="fu">filter_snp</span><span class="op">(</span><span class="va">gene</span>, gene_type <span class="op">=</span> <span class="st">"INDEX"</span>, upstream_length <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span></code></pre></div><p></p></div>
</div>

<div class="section">
<h4 id="arguments-4">Arguments<a class="anchor" aria-label="anchor" href="#arguments-4"></a></h4>
<p></p><div class="arguments"><dl><dt><code>gene</code></dt>
<dd><p>values for Index , Locus_tag or Gene</p></dd>


<dt><code>gene_type</code></dt>
<dd><p>values for searching mutations by certain type, should be one of
"GENE","LOCUS_TAG","INDEX"</p></dd>


<dt><code>upstream_length</code></dt>
<dd><p>whether to show the upstream mutation of genes</p></dd>


<dt><code>length</code></dt>
<dd><p>Length of upstream and downstream regions of genes that should be included
Initialize a new MyClass object</p></dd>


</dl><p></p></div>
</div>

</div><p></p><hr><a id="method-Mutation Object-filter_sample"></a><div class="section">
<h3 id="method-filter-sample-">Method <code>filter_sample()</code><a class="anchor" aria-label="anchor" href="#method-filter-sample-"></a></h3>

<div class="section">
<h4 id="usage-5">Usage<a class="anchor" aria-label="anchor" href="#usage-5"></a></h4>
<p></p><div class="r"><div class="sourceCode"><pre><code><span><span class="va">SNP</span><span class="op">$</span><span class="fu">filter_sample</span><span class="op">(</span>sample_name <span class="op">=</span> <span class="cn">NULL</span>, mutation_index <span class="op">=</span> <span class="cn">NULL</span>, opposite <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></span></code></pre></div><p></p></div>
</div>

<div class="section">
<h4 id="arguments-5">Arguments<a class="anchor" aria-label="anchor" href="#arguments-5"></a></h4>
<p></p><div class="arguments"><dl><dt><code>sample_name</code></dt>
<dd><p>sample names to filter</p></dd>


<dt><code>mutation_index</code></dt>
<dd><p>mutation index.</p></dd>


<dt><code>opposite</code></dt>
<dd><p>Used with mutation_index. When opposite is set to TRUE, samples
containing mutation_index will be removed; otherwise will be remained.
Getting SNPs counts in all samples</p></dd>


</dl><p></p></div>
</div>

</div><p></p><hr><a id="method-Mutation Object-estimate_imp_all"></a><div class="section">
<h3 id="method-estimate-imp-all-">Method <code>estimate_imp_all()</code><a class="anchor" aria-label="anchor" href="#method-estimate-imp-all-"></a></h3>

<div class="section">
<h4 id="usage-6">Usage<a class="anchor" aria-label="anchor" href="#usage-6"></a></h4>
<p></p><div class="r"><div class="sourceCode"><pre><code><span><span class="va">SNP</span><span class="op">$</span><span class="fu">estimate_imp_all</span><span class="op">(</span><span class="op">)</span></span></code></pre></div><p></p></div>
</div>


</div><p></p><hr><a id="method-Mutation Object-estimate_imp"></a><div class="section">
<h3 id="method-estimate-imp-">Method <code>estimate_imp()</code><a class="anchor" aria-label="anchor" href="#method-estimate-imp-"></a></h3>

<div class="section">
<h4 id="usage-7">Usage<a class="anchor" aria-label="anchor" href="#usage-7"></a></h4>
<p></p><div class="r"><div class="sourceCode"><pre><code><span><span class="va">SNP</span><span class="op">$</span><span class="fu">estimate_imp</span><span class="op">(</span></span>
<span>  <span class="va">gene</span>,</span>
<span>  gene_type <span class="op">=</span> <span class="st">"GENE"</span>,</span>
<span>  metadata_col <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  upstream_length <span class="op">=</span> <span class="cn">NULL</span></span>
<span><span class="op">)</span></span></code></pre></div><p></p></div>
</div>

<div class="section">
<h4 id="arguments-6">Arguments<a class="anchor" aria-label="anchor" href="#arguments-6"></a></h4>
<p></p><div class="arguments"><dl><dt><code>gene</code></dt>
<dd><p>values for Index , Locus_tag or Gene</p></dd>


<dt><code>gene_type</code></dt>
<dd><p>values for searching mutations by certain type, should be one of
"GENE","LOCUS_TAG","INDEX"</p></dd>


<dt><code>metadata_col</code></dt>
<dd><p>Default is NULL. If the metadata_col was set, the estimate_imp function
will calculate significance difference of mutation counts among different metadata_col levels.
Should be colnames in metadata or the mutation INDEX.</p></dd>


<dt><code>upstream_length</code></dt>
<dd><p>whether to show the upstream mutation of genes
MutTree
plot phylogenetic tree with heatmap of snps and lineage composition</p></dd>


</dl><p></p></div>
</div>

</div><p></p><hr><a id="method-Mutation Object-MutTree"></a><div class="section">
<h3 id="method-muttree-">Method <code>MutTree()</code><a class="anchor" aria-label="anchor" href="#method-muttree-"></a></h3>

<div class="section">
<h4 id="usage-8">Usage<a class="anchor" aria-label="anchor" href="#usage-8"></a></h4>
<p></p><div class="r"><div class="sourceCode"><pre><code><span><span class="va">SNP</span><span class="op">$</span><span class="fu">MutTree</span><span class="op">(</span><span class="va">locus_tag</span>, limit <span class="op">=</span> <span class="fl">5</span>, size <span class="op">=</span> <span class="fl">20</span>, metadata_col <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span></code></pre></div><p></p></div>
</div>

<div class="section">
<h4 id="arguments-7">Arguments<a class="anchor" aria-label="anchor" href="#arguments-7"></a></h4>
<p></p><div class="arguments"><dl><dt><code>locus_tag</code></dt>
<dd><p>the locus_tag of wanted gene</p></dd>


<dt><code>limit</code></dt>
<dd><p>the minimum frequence a sample need to have,defaults to 5</p></dd>


<dt><code>size</code></dt>
<dd><p>the number of top occured mutation types to plot,defaults to 20</p></dd>


<dt><code>metadata_col</code></dt>
<dd><p>Default is NULL. If the metadata_col was set, the estimate_imp function
will calculate significance difference of mutation counts among different metadata_col levels.
Should be colnames in metadata or the mutation INDEX.
MutNumPlot
plot mutation times and modification types in a single gene.</p></dd>


</dl><p></p></div>
</div>

</div><p></p><hr><a id="method-Mutation Object-MutNumPlot"></a><div class="section">
<h3 id="method-mutnumplot-">Method <code>MutNumPlot()</code><a class="anchor" aria-label="anchor" href="#method-mutnumplot-"></a></h3>

<div class="section">
<h4 id="usage-9">Usage<a class="anchor" aria-label="anchor" href="#usage-9"></a></h4>
<p></p><div class="r"><div class="sourceCode"><pre><code><span><span class="va">SNP</span><span class="op">$</span><span class="fu">MutNumPlot</span><span class="op">(</span></span>
<span>  <span class="va">locus_tag</span>,</span>
<span>  limit <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  label_num <span class="op">=</span> <span class="fl">10</span>,</span>
<span>  show_synomonous <span class="op">=</span> <span class="cn">T</span>,</span>
<span>  Modification <span class="op">=</span> <span class="cn">NULL</span></span>
<span><span class="op">)</span></span></code></pre></div><p></p></div>
</div>

<div class="section">
<h4 id="arguments-8">Arguments<a class="anchor" aria-label="anchor" href="#arguments-8"></a></h4>
<p></p><div class="arguments"><dl><dt><code>locus_tag</code></dt>
<dd><p>the locus_tag of wanted gene</p></dd>


<dt><code>limit</code></dt>
<dd><p>the minimum frequence a sample need to have,defaults to 1</p></dd>


<dt><code>label_num</code></dt>
<dd><p>the number of top frequent mutationsto label, defaults to 10</p></dd>


<dt><code>show_synomonous</code></dt>
<dd><p>Whether to show synomonous mutations.</p></dd>


<dt><code>Modification</code></dt>
<dd><p>Modification types to show.
calculate_NJ
calculate sample distance for object based on Neighbor-Joining Algorithm.</p></dd>


<dt><code>metadata_col</code></dt>
<dd><p>Default is NULL. If the metadata_col was set, the estimate_imp function</p></dd>


</dl><p></p></div>
</div>

</div><p></p><hr><a id="method-Mutation Object-calculate_NJ"></a><div class="section">
<h3 id="method-calculate-nj-">Method <code>calculate_NJ()</code><a class="anchor" aria-label="anchor" href="#method-calculate-nj-"></a></h3>

<div class="section">
<h4 id="usage-10">Usage<a class="anchor" aria-label="anchor" href="#usage-10"></a></h4>
<p></p><div class="r"><div class="sourceCode"><pre><code><span><span class="va">SNP</span><span class="op">$</span><span class="fu">calculate_NJ</span><span class="op">(</span>num_thread <span class="op">=</span> <span class="fl">1</span>, snp_range <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.05</span>, <span class="fl">0.95</span><span class="op">)</span><span class="op">)</span></span></code></pre></div><p></p></div>
</div>

<div class="section">
<h4 id="arguments-9">Arguments<a class="anchor" aria-label="anchor" href="#arguments-9"></a></h4>
<p></p><div class="arguments"><dl><dt><code>num_thread</code></dt>
<dd><p>number of threads to use</p></dd>


<dt><code>snp_range</code></dt>
<dd><p>SNPs within this frequency range were included in the NJ (Neighbor-Joining) distance calculation. This range was used to accelerate computational speed.</p></dd>


</dl><p></p></div>
</div>

</div><p></p><hr><a id="method-Mutation Object-clone"></a><div class="section">
<h3 id="method-clone-">Method <code>clone()</code><a class="anchor" aria-label="anchor" href="#method-clone-"></a></h3>
<p>The objects of this class are cloneable with this method.</p><div class="section">
<h4 id="usage-11">Usage<a class="anchor" aria-label="anchor" href="#usage-11"></a></h4>
<p></p><div class="r"><div class="sourceCode"><pre><code><span><span class="va">SNP</span><span class="op">$</span><span class="fu">clone</span><span class="op">(</span>deep <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div><p></p></div>
</div>

<div class="section">
<h4 id="arguments-10">Arguments<a class="anchor" aria-label="anchor" href="#arguments-10"></a></h4>
<p></p><div class="arguments"><dl><dt><code>deep</code></dt>
<dd><p>Whether to make a deep clone.</p></dd>


</dl><p></p></div>
</div>

</div>

    </div>

  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Hongxiang Xu.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer></div>

  

  

  </body></html>

